src/utils/PolygonConnector.h

```diff
@@ -166,7 +489,88 @@ class PolygonConnector
      * - the first connection at a whole line distance away
      * So as to try and find a bridge which is centered around the initiall found first connection
      */
-    std::optional<PolygonBridge> getBridge(ConstPolygonRef poly, std::vector<Polygon>& polygons);
+    template<typename Polygonal>
+    std::optional<PolygonBridge<Polygonal>> getBridge(Polygonal& from_poly, std::vector<Polygonal>& to_polygons)
+    {
+        std::optional<PolygonBridge<Polygonal>> connection = findConnection(from_poly, to_polygons);
+        if(!connection) //We didn't find a connection. No bridge.
+        {
+            return std::nullopt;
+        }
+
+        //Ensure that B is always the right connection and A the left.
+        if(LinearAlg2D::pointIsLeftOfLine(connection->b.from_point, connection->a.from_point, connection->a.to_point) > 0)
+        {
+            std::swap(connection->a, connection->b);
+        }
+        return connection;
+    }
+
+    /*!
+     * Walk along a polygon to find the first point that is exactly ``distance``
+     * away from a given line.
+     *
+     * The resulting point does not have to be exactly on a vertex. Most likely
+     * it will be on a line segment.
+     * \param poly The polygonal shape along which to walk.
+     * \param start_index The vertex at which to start looking. This vertex
+     * should be on the wrong side of the line.
+     * \param distance The distance from the line at which the resulting point
+     * should be.
+     * \param line_a The line passes through this point.
+     * \param line_b The line also passes through this point.
+     * \param direction Use +1 to iterate in the forward direction through the
+     * polygon, or -1 to iterate backwards.
+     * \return If there is a point that is the correct distance from the line,
+     * the first such point is returned, and the segment index that it's on. If
+     * the polygon is entirely close to the line, returns
+     * ``std::nullopt``.
+     */
+    template<typename Polygonal>
+    std::optional<std::pair<Point, size_t>> walkUntilDistanceFromLine(const Polygonal& poly, const size_t start_index, const coord_t distance, const Point& line_a, const Point& line_b, const short direction)
+    {
+        const size_t poly_size = poly.size();
+        const coord_t line_magnitude = vSize(line_b - line_a); //Pre-compute, used for line distance calculation.
+        if(line_magnitude == 0)
+        {
+            return std::nullopt; //Line doesn't have a direction, so we can't be on any one side of it.
+        }
+
+        for(size_t index = (start_index + direction + poly_size) % poly_size; index != start_index; index = (index + direction + poly_size) % poly_size)
```

Comment:
The danger of off-by-one errors in this sort of loops is real. If it can be explained better, feel free to correct it! I'm almost inclined to do the ASCII-art thing but that is also a lot of work.

Reply:
It's fine, out of all the remarks, this is the most 'just a comment' one.